core/storage/platform_parameter/gae_models.py

```diff
@@ -0,0 +1,70 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models relating to configuration properties."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import core.storage.base_model.gae_models as base_models
+
+from google.appengine.ext import ndb
+
+
+class PlatformParameterSnapshotMetadataModel(
+        base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a platform parameter snapshot."""
+
+    pass
+
+
+class PlatformParameterSnapshotContentModel(
+        base_models.BaseSnapshotContentModel):
+    """Storage model for the content for a platform parameter snapshot."""
+
+    pass
+
+
+class PlatformParameterModel(base_models.VersionedModel):
+    """A class that represents a named dynamic platform parameter.
+    This model only stores fields that can be updated in run time.
+
+    The id is the name of the parameter.
```

Comment:
It seems worthwhile to have a test to verify that reusing parameter names introduces an error. Note to you: we should also verify at the domain level that parameter IDs remain unique forever (even after the feature is launched & finished).

Reply:
Looks like that saving two Model instances with the same id does not raise an error, the older one is simply overwritten by the later one. So I think we don't need a test case for this since it's the original behavior of ndb.Models and I did not add extra logics on this.
There are logics and tests preventing from reusing parameter names at the domain level.

Reply:
Ack--thanks for clarifying (and making sure that this is enforced at the domain level).